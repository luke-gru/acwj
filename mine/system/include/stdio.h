#ifndef _STDIO_H_
#define _STDIO_H_

#include <stddef.h>
#include <stdarg.h>

#define EOF (-1)

// This FILE definition will do for now
typedef char *FILE;
extern FILE *stderr;
extern FILE *stdout;

FILE *fopen(char *pathname, char *mode);
size_t fread(void *ptr, size_t size, size_t nmemb, FILE *stream);
size_t fwrite(void *ptr, size_t size, size_t nmemb, FILE *stream);
int fclose(FILE *stream);
int printf(char *format);
int fprintf(FILE *stream, char *format);
size_t getline(char **lineptr, size_t *n, FILE *stream);

FILE *popen(const char *command, const char *type);
int pclose(FILE *stream);

int vfprintf(FILE *stream, const char *format, va_list ap);

#endif	// _STDIO_H_
