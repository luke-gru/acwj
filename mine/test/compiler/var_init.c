#include <stdio.h>

int f;
int g = 1;
int a[10];                                      // Ten zeroed elements
char b[]= { 'q', 'w', 'e', 'r', 't', 'y' };     // Six elements
char c[10]= { 'q', 'w', 'e', 'r', 't', 'y' };   // Ten elements, zero padded
char *hi = "hello";

int main() {
  printf("f: %d\n", f);
  printf("g: %d\n", g);
  printf("a[0]: %d\n", a[0]);
  printf("b[0]: %c\n", b[0]);
  printf("c[2]: %c\n", c[2]);
  printf("%s\n", hi);
  return (0);
}
