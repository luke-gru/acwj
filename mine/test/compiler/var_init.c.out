f: 0
g: 1
a[0]: 0
b[0]: q
c[2]: e
hello
